### PR TITLE
Adds the capacity to add properties to particle generators that use the dofs support points or the quadrature points

### PR DESCRIPTION
**Pull request by blaisb**
Added the capacity to add properties to the particles that are generated using the particle generator that generates particles at support points or at gauss quadrature points.
Note that we do not check for the size of the properties and the number of particles
as this is done within insert global particles. This is a very simple pull request, but it will be required for step-68 and could actually be used to simplify step-70. Comments are more than welcomed.

@gassmoeller , this fixes the issue I had with adding the particle properties to step-68

### CODE REVIEW

File: include/deal.II/particles/generators.h [hunk 1]
```diff
@@ -207,6 +210,8 @@ namespace Particles
                        ParticleHandler<dim, spacedim> &particle_handler,
                        const Mapping<dim, spacedim> &  mapping =
                          StaticMappingQ1<dim, spacedim>::mapping,
+                       const std::vector<std::vector<double>> &properties =
```
Comment (peterrum) [inline include/deal.II/particles/generators.h:213]: Could you switch the order of `properties` and `components`. I think we need that for backwards  compatibility.

**Comment by blaisb** [comment]
@peterrum You are right :). I fixed that. I also found a mistake I had made in the call to insert_global_particles()

File: include/deal.II/particles/generators.h [hunk 2]
```diff
@@ -194,6 +194,9 @@ namespace Particles
      * @param[in] mapping An optional mapping object that is used to map
      * the DOF locations. If no mapping is provided a MappingQ1 is assumed.
      *
+     * @param[in] properties An optional vector of vector of properties
```
Comment (peterrum) [inline include/deal.II/particles/generators.h:197]: I guess this comment is misplaced now.

File: include/deal.II/particles/generators.h [hunk 3]
```diff
@@ -194,6 +194,9 @@ namespace Particles
...
+     * for each particles to be inserted.
```
Comment (peterrum) [inline include/deal.II/particles/generators.h:198]:
```suggestion
     * for each particle to be inserted.
```

File: include/deal.II/particles/generators.h [hunk 4]
```diff
@@ -237,6 +242,9 @@ namespace Particles
      * the quadrature locations. If no mapping is provided a MappingQ1 is
      * assumed.
      *
+     * @param[in] properties An optional vector of vector of properties
+     * for each particles to be inserted.
```
Comment (peterrum) [inline include/deal.II/particles/generators.h:246]:
```suggestion
     * for each particle to be inserted.
```

File: include/deal.II/particles/generators.h [hunk 5]
```diff
@@ -207,7 +210,9 @@ namespace Particles
                        ParticleHandler<dim, spacedim> &particle_handler,
                        const Mapping<dim, spacedim> &  mapping =
                          StaticMappingQ1<dim, spacedim>::mapping,
-                       const ComponentMask &components = ComponentMask());
+                       const ComponentMask &components = ComponentMask(),
+                       const std::vector<std::vector<double>> &properties =
+                         std::vector<std::vector<double>>());
```
Comment (peterrum) [inline include/deal.II/particles/generators.h:215]:
```suggestion
                         {});
```
as done in `insert_global_particles`.

**Comment by blaisb** [comment]
This addresses your comments :). Thank you so much for the review Peter.

**Comment by peterrum** [comment]
@gassmoeller Are fine if I merge?

**Comment by gassmoeller** [comment]
Yes this looks good to me.

**Pull request merged**